### PR TITLE
FS Dashboard

### PR DESCRIPTION
**Pull request by rajivtitus**
### What this PR does (required):
- Changed sign-up form to include first and last name - creates profile on user creation (removed username)
- Fetches all sitter profiles when user is not logged in. (isDogSitter marked to true and have price and city)
- When the user is logged in, fetches all sitter profiles excluding the current user
- You can click on a profile and route to profile details page
- Navbar now displays profile photo
- Added grow, fade animations and loading spinners to profile details and listing pages

Note: Currently, the search bar feature in the dashboard and the send request form in the profile details are not yet functional.
Also, ratings for sitter profiles have been hardcoded at the moment, I'm almost done working on the controllers to fetch reviews. Post that, ratings and reviews will be fetched from the API and only logged in users will be able to rate and create reviews.

### Any information needed to test this feature (required):
- `npm start` on client | `npm run dev` on server
- Please make sure you are logged out
- Enter /listings in url to view all dog sitter profiles. Eg: `http://localhost:3000/listings`
- Click on listing to route to profile details
- Create an account or login using existing account, you will be taken to the listing page.

### Any issues with the current functionality (optional):
- None

### CODE REVIEW

File: client/src/components/Listings/ProfileList/ProfileList.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  if (loading) return <LoadingSpinner />;
+
+  if (errorMsg) updateSnackBarMessage(errorMsg);
+
+  return (
+    <Grow in={true}>
+      <Grid container spacing={8}>
+        {userProfiles.map((userProfile, index) => (
+          <Grid item xs={12} sm={4} key={index}>
```
Comment (sina-jamshidi) [inline client/src/components/Listings/ProfileList/ProfileList.tsx]: if we have userProfile, do we also have some other identifier like id that we can use as key?

File: client/src/components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ interface Props {
 export default function Navbar({ user, logout }: Props): JSX.Element {
   const classes = useStyles();
   const history = useHistory();
+  console.log('render');
```
Comment (sina-jamshidi) [inline client/src/components/Navbar/Navbar.tsx]: remove

File: server/controllers/profile.js [hunk 3]
```diff
@@ -1,47 +1,50 @@
...
+    const userProfiles = users.filter(user => user.profile != null);
+    res.status(200).json({
+      success: {
+        users: userProfiles,
       }
+    });
+  } catch (error) {
+    res.status(500);
+    throw new Error(error.message);
+  }
```
Comment (sina-jamshidi) [inline server/controllers/profile.js]: this is overkill. We can just hit the regular route to fetch users, and check if `req.user.id` exists. If it does, exclude it

File: client/src/helpers/APICalls/searchProtectedProfiles.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    method: 'GET',
+    headers: { 'Content-Type': 'application/json' },
+    credentials: 'include',
+  };
+  return await fetch(`/profiles/protected`, fetchOptions)
+    .then((res) => res.json())
+    .catch(() => ({
+      error: { message: 'Unable to connect to server. Please try again' },
+    }));
+};
```
Comment (sina-jamshidi) [inline client/src/helpers/APICalls/searchProtectedProfiles.ts]: I don't think this is necessary, see comment below

**Review by sina-jamshidi — APPROVED** [review]
Looks good! Just a few comments to address

File: client/src/components/Listings/ProfileList/ProfileList.tsx (hunk 1 above)
Comment (rajivtitus) [reply client/src/components/Listings/ProfileList/ProfileList.tsx]: I must've failed to notice that I was using an index. I have changed to the profile's ID now. Thanks!

File: client/src/components/Navbar/Navbar.tsx (hunk 2 above)
Comment (rajivtitus) [reply client/src/components/Navbar/Navbar.tsx]: Done

File: client/src/helpers/APICalls/searchProtectedProfiles.ts (hunk 4 above)
Comment (rajivtitus) [reply client/src/helpers/APICalls/searchProtectedProfiles.ts]: Answered above.

File: server/controllers/profile.js (hunk 3 above)
Comment (rajivtitus) [reply server/controllers/profile.js]: Initially, I was trying to use one route but on excluding the `protect` middleware (to make the route accessible to all users), the JWT token from the FE is not decoded as a result `req.user = undefined` on all requests. Hence, I created two routes.

Currently, I have conditionally applied the `protect` middleware, see below:

Capture

Thanks for pointing this out! :)

File: client/src/interface/Profile.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      end: Date;
+    },
+  ];
+  price: number;
+  city: string;
+  gallery: [string];
+  profilePhoto?: string;
+  coverPhoto?: string;
+  description?: string;
+  phoneNumber?: Number;
```
Comment (xia0m) [inline client/src/interface/Profile.ts:18]: should the phone number be type string?, also, it should be lower case number

File: client/src/interface/Profile.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  address?: String;
```
Comment (xia0m) [inline client/src/interface/Profile.ts:19]: should be string here

**Pull request merged**